src/joy_api.c

```diff
@@ -565,6 +590,148 @@ int joy_initialize(joy_init_t *init_data,
     return ok;
 }
 
+/*
+ * Function: joy_initialize_no_config
+ *
+ * Description: This function initializes the Joy library
+ *      to analyze the data features. This fucntion does not
+ *      perform any configuration options as it is expetced that
+ *      the caller did configuration prior to calling this function.
+ *
+ *      joy_initialize must be called before using any of the other
+ *      API functions.
+ *
+ * Parameters:
+ *      config - pointer to pre-setup config
+ *      err_info - pointer to the file for error logging
+ *      data - structure of Joy options
+ *
+ * Returns:
+ *      0 - success
+ *      1 - failure
+ *
+ */
+int joy_initialize_no_config(void *config, FILE *err_info, joy_init_t *data)
+{
+    unsigned int i = 0;
+    int dir_len = 0;
+    char output_filename[MAX_FILENAME_LEN];
+
+    /* must pass in a pre-setup configuartion */
+    if (config == NULL) {
+        joy_log_err("could not initialize the protocol identification dictionary");
+        return failure;
+    }
+
+    /* make sure logfile/stderr is setup */
+    glb_config = (configuration_t*)config;
+    if (err_info == NULL) {
+        info = stderr;
+    } else {
+        info = err_info;
+    }
+
+    /* sanity check the context information */
+    if (data->contexts < 1) {
+        data->contexts = 1; /* default to 1 context thread */
+    }
+
+    /* allocate the context memory */
+    JOY_API_ALLOC_CONTEXT(ctx_data, data->contexts)
+    joy_num_contexts = data->contexts;
+
+    glb_config->flow_key_match_method = EXACT_MATCH;
+
+    /* setup the inactive and active timeouts for a flow record */
+    flow_record_update_timeouts(data->inact_timeout, data->act_timeout);
+
+    /* initialize the protocol identification dictionary */
+    if (proto_identify_init()) {
+        joy_log_err("could not initialize the protocol identification dictionary");
+        JOY_API_FREE_CONTEXT(ctx_data);
+        return failure;
+    }
+
+    /* setup the output directory */
+    if (glb_config->outputdir == NULL) {
+        glb_config->outputdir = strdup("./");
+    }
+    dir_len = strlen(glb_config->outputdir);
+    if (glb_config->outputdir[dir_len-1] == '/') {
+        glb_config->outputdir[dir_len-1] = '\0';
+    }
+
+    /* initialize all the data context structures */
+    for (i=0; i < JOY_MAX_CTX_INDEX(ctx_data); ++i) {
+        struct joy_ctx_data *this = JOY_CTX_AT_INDEX(ctx_data,i);
```

Comment:
joy_ctx_data should have a typedef joy_ctx_data_t

Reply:
its a circular definition so that actual implementation is hidden from API users in the private file.